### PR TITLE
Fixes for mob spawners

### PR DESCRIPTION
**Pull request by CheffieGithub**
## About the PR 
Fixes death alert spam and laysound sounds. Adds puritan to stop revival. Adds a variable for PDAs which can be set to stop the device showing on scans to prevent the list getting clogged by corpses.

## Why's this needed? 
10x owlery death alerts not good nor is the pda list getting filled with dead PDAs.

### CODE REVIEW

File: code/obj/mob_spawn.dm [hunk 1]
```diff
@@ -95,8 +95,14 @@
 		if (H.r_hand)
 			qdel(H.r_hand)
 
+		SPAWN(1)
+			for (var/obj/item/implant/health/implant as anything in H.implant)
+				qdel(implant)
```
Comment (TobleroneSwordfish) [inline code/obj/mob_spawn.dm:100]:
```suggestion
				qdel(implant)
			H.implant = list()
```
best not to leave references to qdel'd things lying around

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

**Comment by Tarmunora** [comment]
I think the corpses' PDAs might be showing up when scanning for PDAs - probably want to fix that one way or another

**Comment by CheffieGithub** [comment]
I have an idea for how to fix the scanning which could also be extended to salvagers and the ops pda, probably

**Comment by CheffieGithub** [comment]
Added a variable for PDAs to indicate scannablity if this is a good solution it should be added to pda2/salvager and pda2/syndicate I think

**Pull request merged**